kernel/src/syscall.rs

```diff
@@ -25,28 +93,45 @@ pub enum Syscall {
 
     /// Instruct the kernel or a capsule to perform an operation.
     ///
-    /// SVC_NUM = 2
-    COMMAND {
+    /// System call class ID 2
+    Command {
         driver_number: usize,
         subdriver_number: usize,
         arg0: usize,
         arg1: usize,
     },
 
-    /// Share a memory buffer with the kernel.
+    /// Share a memory buffer with the kernel, which the kernel may
+    /// read from and write to.
     ///
-    /// SVC_NUM = 3
-    ALLOW {
+    /// System call class ID 3
+    ReadWriteAllow {
         driver_number: usize,
         subdriver_number: usize,
         allow_address: *mut u8,
         allow_size: usize,
     },
 
+    /// Share a memory buffer with the kernel, which the kernel may
+    /// only read from and never write to.
+    ///
+    /// System call class ID 4
+    ReadOnlyAllow {
+        driver_number: usize,
+        subdriver_number: usize,
+        allow_address: *const u8,
+        allow_size: usize,
+    },
+
     /// Various memory operations.
     ///
-    /// SVC_NUM = 4
-    MEMOP { operand: usize, arg0: usize },
+    /// System call class ID 5
+    Memop { operand: usize, arg0: usize },
+
+    Exit {
```

Comment:
```suggestion
    /// Process exit, with optional request-to-restart.
    ///
    /// System call class ID 6
    Exit {
```

Reply:
I'm updating this with more useful comments.